frontend/src/modals/upload-document/UploadDocumentModal.tsx

```diff
@@ -64,38 +76,86 @@ export const UploadDocumentModal = ({ projectId, onDocumentUploaded, onClose })
       },
     };
     try {
-      const response = await sendFile(uploadDocumentMutation, variables, file);
-      const action = { type: HANDLE_SUBMIT_ACTION, response };
-      dispatch(action);
+      const { data, error } = await sendFile(uploadDocumentMutationFile, variables, file);
+      if (error) {
+        const showToastEvent: ShowToastEvent = {
+          type: 'SHOW_TOAST',
+          message: error.message,
+        };
+        dispatch(showToastEvent);
+      } else if (data) {
+        const event: HandleResponseEvent = { type: 'HANDLE_RESPONSE', data };
+        dispatch(event);
+
+        const { uploadDocument } = data;
+        if (isErrorPayload(uploadDocument)) {
+          const { message } = uploadDocument;
+          const showToastEvent: ShowToastEvent = { type: 'SHOW_TOAST', message };
+          dispatch(showToastEvent);
+        }
+      }
     } catch (exception) {
       // Handle other errors like max file size error send by the backend...
-      const action = { type: HANDLE_SUBMIT_ACTION, response: { error: exception.toString() } };
-      dispatch(action);
+      const showToastEvent: ShowToastEvent = { type: 'SHOW_TOAST', message: exception.toString() };
+      dispatch(showToastEvent);
     }
   };
 
   const onFileSelected = (file) => {
-    const action = { type: HANDLE_SELECTED_ACTION, file };
-    dispatch(action);
+    const selectModelEvent: SelectModelEvent = { type: 'SELECT_MODEL', file };
+    dispatch(selectModelEvent);
   };
 
   useEffect(() => {
-    if (viewState === SUBMIT_SUCCESS_STATE) {
+    if (uploadModelModal === 'success') {
       onDocumentUploaded();
     }
-  }, [viewState, onDocumentUploaded]);
+  }, [uploadModelModal, onDocumentUploaded]);
```

Comment:
Same here and in the state machine, the modal is named, on purpose, uploadDocumentModal

Reply:
Done